common/test/acceptance/tests/lms/test_lms.py

```diff
@@ -260,6 +273,15 @@ def test_immediate_verification_enrollment(self):
         # Submit photos and proceed to the enrollment confirmation step
         self.payment_and_verification_flow.next_verification_step(self.immediate_verification_page)
 
+        # Expect that one mode_changed enrollment event fired as part of the upgrade
+        assert_event_emitted_num_times(
```

Comment:
Since this event is emitted immediately after the user completes payment and is enrolled in the verified mode, I'd move this up to immediately after `self.fake_payment_page.submit_payment()`.